### DISCUSSION TITLE
FontIcon versus TextBlock (with only an icon code as content)

### DISCUSSION THREAD

**Discussion by Jay-o-Way** [microsoft/microsoft-ui-xaml#8612] [Q&A] [answered]
Sometimes I see **FontIcon** used, and sometimes I see a **TextBlock** where the only content is an icon code. Technically these both work fine, but I have a feeling the FontIcon _should_ be used in any/all of these cases. Is there an official ~rule~ guideline from Microsoft about this?

And if so, we could convert this (item) into an actionable issue, right? I mean, since this applies to the code in this repo just aswel.

**Comment by codendone** [comment, marked as answer]
`FontIcon` exists for use in places which require an `IconElement`, such as for AppBarButton.Icon or MenuFlyoutItem.Icon. It is fine to use `FontIcon` elsewhere, but internally it just uses a `TextBlock` which would be fine to use directly in places which don't require an `IconElement`.
